langchain/src/retrievers/self_query/utils.ts

```diff
@@ -0,0 +1,23 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export function isObject(obj: any): obj is object {
+  return (
+    obj &&
+    typeof obj === "object" &&
+    !Array.isArray(obj) &&
+    obj.constructor.name === "Object"
```

Comment:
I think this will get messed up in minified environments - will the first three checks suffice?

Reply:
Yeah I think it should be fine.